src/coprocessor/dag/rpn_expr/impl_op.rs

```diff
@@ -26,9 +26,41 @@ impl RpnFnLogicalAnd {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct RpnFnLogicalOr;
+
+impl_template_fn! { 2 arg @ RpnFnLogicalOr }
+
+impl RpnFnLogicalOr {
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg0: &Option<i64>,
+        arg1: &Option<i64>,
+    ) -> Result<Option<i64>> {
+        // Rule 1. As long as there is `Some(x)` where x is not 0, the result is `1`.
+        // Rule 2. As long as there is `None`, the result is `None`.
+        // Rule 3. For the rest, the result is `0`.
+        Ok(match arg0 {
+            None => match arg1 {
```

Comment:
The comment is, when rule 1 does not apply, see rule 2. When rule 2 does not apply, see rule 3.
So basically, for example, when lhs is `Some(5)` and `rhs` is `None`, the result is not `None` but `Some(1)` because it follows rule 1 and is finished.

Reply:
It would make sense to me to say

```
A nullable logical OR with a mapping from Rust to logic of

* None => null
* Some(0) => false
* Some(x != 0) => true

Here a null always propagates as the result of a logical OR:
null OR x == null
```

Note that this treatment of null is different than [MySQL](https://dev.mysql.com/doc/refman/8.0/en/logical-operators.html). MySQL is using the Kleene logic used in the SQL standard where `null OR true == true`.


You could also remove repetition here by writing the code that way (re-use a single mapping function twice), but the code might be a little slower.

Reply:
Oh I mis-read the code. `null OR true == true`

Reply:
The main logic looks good to me, can you refine the comments according to the suggestion of gregwebs?